### PR TITLE
feat/redesign chat list

### PR DESCRIPTION
**Pull request by sieuhuflit**
## Overview

Redesign whole chatlist screen
- Increase size of profile image
- Add created at time of message
- Show latest message
- Merged styling of direct message & channel message
- Show animation typing also
- Show channel avatar
- Show online/offline status
- Handle decrypted message and show message correctly

## Dependencies added

 flutter_olm
 flutter_openssl_crypto

### CODE REVIEW

File: lib/core/presentation/widgets/chat/matrix_avatar.dart [hunk 1]
```diff
@@ -48,33 +54,93 @@ class MatrixAvatar extends StatelessWidget {
...
-                width: size,
-                height: size,
-                placeholder: (_) => textWidget,
-                cacheKey: mxContent.toString(),
-              ),
-      ),
+    final borderRadius = BorderRadius.circular(radius ?? LemonRadius.xSmall);
+    final container = Stack(
+      alignment: Alignment.center,
+      children: [
```
Comment (roncoderpush) [inline lib/core/presentation/widgets/chat/matrix_avatar.dart:60]: - Avoid putting too many variables in build method. 
- For rich widget such as `container` here I suggest we put it in separate class. 
- We don't need an if else statement whether onTap or null.

File: lib/core/utils/chat/room_status_extension.dart [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+import 'date_time_extension.dart';
+
+extension RoomStatusExtension on Room {
+  CachedPresence? get directChatPresence =>
+      client.presences[directChatMatrixID];
+
+  String getLocalizedStatus(BuildContext context) {
+    final t = Translations.of(context);
+    if (isDirectChat) {
+      final directChatPresence = this.directChatPresence;
```
Comment (roncoderpush) [inline lib/core/utils/chat/room_status_extension.dart:17]: Complex logic, prefer put an explanation here

**Review by roncoderpush — CHANGES_REQUESTED** [review]

File: lib/core/utils/chat/room_status_extension.dart (hunk 2 above)
Comment (sieuhuflit) [reply lib/core/utils/chat/room_status_extension.dart:17]: Update some comment there

File: lib/core/presentation/widgets/chat/matrix_avatar.dart (hunk 1 above)
Comment (sieuhuflit) [reply lib/core/presentation/widgets/chat/matrix_avatar.dart:60]: > We don't need an if else statement whether onTap or null.

Thanks, remove the weird condition here

> Avoid putting too many variables in build method.
> For rich widget such as container here I suggest we put it in separate class.

This widget not so big, it's just render MatrixAvatar only

**Review by mthinh — APPROVED** [review]

**Review by mthinh — APPROVED** [review]

**Pull request merged**